modules/Sources/Dependencies/AddressBookClient/AddressBookEncryption.swift

```diff
@@ -272,15 +244,25 @@ extension AddressBookClient {
     }
     
     // Helper function to read a string from the data using a length prefix
-    private static func readString(from data: Data, at offset: inout Int) -> String? {
+    private static func readString(from data: Data, at offset: inout Int) throws -> String? {
         // Read the length first (assumes the length is stored as an Int)
-        let lengthBytes = data.subdata(in: offset..<(offset + MemoryLayout<Int>.size))
-        offset += MemoryLayout<Int>.size
+        let lengthBytes = try AddressBookClient.subdata(of: data, in: offset..<(offset + Constants.int64Size))
+        offset += Constants.int64Size
         guard let length = AddressBookClient.bytesToInt(Array(lengthBytes)), length > 0 else { return nil }
         
         // Read the string bytes
-        let stringBytes = data.subdata(in: offset..<(offset + length))
+        let stringBytes = try AddressBookClient.subdata(of: data, in: offset..<(offset + length))
         offset += length
         return AddressBookClient.bytesToString(Array(stringBytes))
     }
+    
+    private static func subdata(of data: Data, in range: Range<Data.Index>) throws -> Data {
+        guard data.count >= range.lowerBound.distance(to: range.upperBound) else {
```

Comment:
Blocking. 

```suggestion
        guard data.count >= range.upperBound else {
```
(`upperBound` is exclusive and relative to the start of `data` in this case. It's not necessary for the check to involve `lowerBound`, because `range.upperBound >= range.lowerBound` for any `range`.)

Reply:
Updated